parlai/mturk/core/test/test_db_interactions.py

```diff
@@ -0,0 +1,547 @@
+                worker_id_1, 'fake_id')), 0)
+
+        conversation_id_1 = "CONV_ID_1"
+        conversation_id_2 = "CONV_ID_2"
+
+        db_logger.log_start_onboard(worker_id_1, assignment_id_1)
+        db_logger.log_start_onboard(worker_id_2, assignment_id_2)
+        db_logger.log_start_onboard(worker_id_2, assignment_id_3)
+        db_logger.log_finish_onboard(worker_id_1, assignment_id_1)
+        db_logger.log_finish_onboard(worker_id_2, assignment_id_2)
+        db_logger.log_finish_onboard(worker_id_2, assignment_id_3)
+        db_logger.log_start_task(worker_id_1, assignment_id_1,
+                                 conversation_id_1)
+        db_logger.log_start_task(worker_id_2, assignment_id_2,
+                                 conversation_id_1)
+        db_logger.log_start_task(worker_id_2, assignment_id_3,
+                                 conversation_id_2)
+
+        # Check to see retrieval by conversation
+        pairs_1 = db_logger.get_pairings_for_conversation(conversation_id_1)
+        pairs_2 = db_logger.get_pairings_for_conversation(conversation_id_2)
+        pairs_3 = db_logger.get_pairings_for_conversation('fake_id')
+        pair_1 = db_logger.get_worker_assignment_pairing(
+            worker_id_1, assignment_id_1)
+        pair_2 = db_logger.get_worker_assignment_pairing(
+            worker_id_2, assignment_id_2)
+        pair_3 = db_logger.get_worker_assignment_pairing(
+            worker_id_2, assignment_id_3)
+        self.assertDictEqual(pairs_1[0], pair_1)
+        self.assertDictEqual(pairs_1[1], pair_2)
+        self.assertDictEqual(pairs_2[0], pair_3)
+        self.assertEqual(len(pairs_3), 0)
+
+        # Do some final processing on assignments
+        db_logger.log_complete_assignment(
+            worker_id_1, assignment_id_1, time.time(),
+            AssignState.STATUS_PARTNER_DISCONNECT)
+        db_logger.log_abandon_assignment(
+            worker_id_2, assignment_id_2, time.time(),
+            AssignState.STATUS_DISCONNECT)
+        db_logger.log_complete_assignment(worker_id_2, assignment_id_3,
+                                          time.time(), AssignState.STATUS_DONE)
+
+        # Assignment state consistent
+        assignment_1_data = db_logger.get_assignment_data(assignment_id_1)
+        assignment_2_data = db_logger.get_assignment_data(assignment_id_2)
+        assignment_3_data = db_logger.get_assignment_data(assignment_id_3)
+        self.assertEqual(assignment_1_data['assignment_id'], assignment_id_1)
+        self.assertEqual(assignment_1_data['status'], 'Reviewable')
+        self.assertIsNotNone(assignment_1_data['approve_time'])
+        self.assertEqual(assignment_1_data['worker_id'], worker_id_1)
+        self.assertEqual(assignment_1_data['hit_id'], HIT1['HIT']['HITId'])
+        self.assertEqual(assignment_2_data['assignment_id'], assignment_id_2)
+        self.assertEqual(assignment_2_data['status'], 'Reviewable')
```

Comment:
so are abandoned HITs now reviewable?

Reply:
In the long run yes, but in the short term not yet. I suppose I should do that transition when the time comes (basic idea is the only HIT that will not be Reviewable is one that was explicitly abandoned, but disconnects can still submit HITs because we'll have bonus per interaction).